dulwich/tests/test_repository.py

```diff
@@ -521,6 +521,23 @@ def check(repo):
             check(nonbare)
             check(bare)
 
+    def test_working_tree(self):
+        temp_dir = tempfile.mkdtemp()
```

Comment:
I do not understand how to do that. Are there other tests doing something similar, so that I can understand what to do?